### PR TITLE
Fixed Burn Cards Animations

### PR DESCRIPTION
**Pull request by DBmaoha**
Burn cards no longer taken right after  using

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 1]
```diff
@@ -69,8 +69,17 @@ int function PlayerInventory_CountBurnRef( entity player, string burnRef )
 void function PlayerInventory_TakeInventoryItem( entity player )
 {
 	entity preexistingWeapon = player.GetOffhandWeapon( OFFHAND_INVENTORY )
-	if ( IsValid( preexistingWeapon ) )
-		player.TakeWeaponNow( preexistingWeapon.GetWeaponClassName() )
+
+	if( !IsValid( preexistingWeapon ) )
```
Comment (BobTheBob9) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:73]: formatting here needs some work, couple newlines around it and it should be good though

**Review by BobTheBob9 — CHANGES_REQUESTED** [review]

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 2]
```diff
@@ -69,8 +69,16 @@ int function PlayerInventory_CountBurnRef( entity player, string burnRef )
...
 	entity preexistingWeapon = player.GetOffhandWeapon( OFFHAND_INVENTORY )
-	if ( IsValid( preexistingWeapon ) )
-		player.TakeWeaponNow( preexistingWeapon.GetWeaponClassName() )
+	if( !IsValid( preexistingWeapon ) )
+		return
+	preexistingWeapon.EndSignal( "OnDestroy" )
+	if( preexistingWeapon.GetWeaponClassName() == "mp_ability_burncardweapon" )
+	{
+		var fireTime = preexistingWeapon.GetWeaponInfoFileKeyField( "fire_anim_rate" )
+		if( fireTime != null )
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:78]:
```suggestion
		if( fireTime )
```

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 3]
```diff
@@ -117,9 +125,9 @@ void function PlayerInventory_PopInventoryItem( entity player )
 
 		if (playerInventoryStack.len() > 0) {
 			InventoryItem nextInventoryItem = playerInventoryStack[playerInventoryStack.len() - 1]
-			PlayerInventory_GiveInventoryItem(player, nextInventoryItem)
+			thread PlayerInventory_GiveInventoryItem(player, nextInventoryItem)
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:128]:
```suggestion
			thread PlayerInventory_GiveInventoryItem( player, nextInventoryItem )
```

**Review by uniboi — CHANGES_REQUESTED** [review]
Rest looks good

**Review by BobTheBob9 — APPROVED** [review]

**Pull request merged**